### ISSUE TITLE
Running `DoMatchParen` upon `BufEnter` causes `:FzfLua git_files` put the selected buffer on a different window.

### ISSUE THREAD

**Issue by doronbehar** [ibhagwan/fzf-lua#768]
### Info

- Operating System: NixOS
- Shell: ZSH
- Terminal: Alacritty
- `nvim --version`: 0.9.0
- `fzf --version`: 0.40.0

- [x] The issue is reproducible with `mini.sh`

But still, it'd be easier to work with the following `init.vim`:

```vim
autocmd BufEnter * execute('DoMatchParen')
```

### Description 

Here's a screencast:

[![asciicast](https://asciinema.org/a/Frfmq3pcP3J8CddMxjgDu7VZi.svg)](https://asciinema.org/a/Frfmq3pcP3J8CddMxjgDu7VZi)

### Context

Continuing the conversation from #646 - apparently this is not an issue specific to `:FzfLua buffers`, but in fact to every `:FzfLua` command that opens a file.

**Comment by doronbehar** [comment]
The source of that `autocmd`, is this:

```vim
autocmd BufEnter * if getfsize(expand("<afile>")) > 1024 * 1024 |
	\ execute('NoMatchParen') |
	\ else |
	\ execute('DoMatchParen') |
\ endif
```

Ideally, Vim / Neovim would have provided the option to disable MatchParen via a buffer local variable, and not a command that affects this globally... Perhaps this should be reported in Neovim's tracker. Putting here the definition of `s:DoMatchParen` for reference:

```vim

func s:DoMatchParen()
  runtime plugin/matchparen.vim
  let w = winnr()
  silent windo doau CursorMoved
  exe "noau ". w . "wincmd w"
endfunc
```

**Comment by ibhagwan** [comment]
> apparently this is not an issue specific to `:FzfLua buffers`, but in fact to every :FzfLua command that opens a file.

Btw, this is not related to opening files, the `DoMatchParen` triggers when opening the main fzf terminal window and changes the active window (not sure why, likely an upstream issue), you can verify that by closing the fzf-lua interface without opening any file and you'll notice the active window also changes to the last opened win.

I'm not sure execatly when that happens or why, fzf-lua's source buffer and window do seem to be accurately stored in:
https://github.com/ibhagwan/fzf-lua/blob/3d6ce0e5bdca5c2e1080255966d52077183b022f/lua/fzf-lua/win.lua#L768-L769

While I could hack my solution by forcing a window change to the source window upon closing of the main interface, this would be an ugly hack with probably other unintended consequences therefore I'd rather this is solved upstream or via a config exclusion of the autocmd.

**Comment by ibhagwan** [comment]
@doronbehar, I was able to solve it by ignoring autocmds only at the initial open, solve this issue (and potentially others) and doesn't prevent the other crucial autocmds from triggering.

Update your plugin to https://github.com/ibhagwan/fzf-lua/commit/0babfa852ed503e66cb11970e2a0aa4b821e9cef and your issue will be solved.

**Comment by doronbehar** [comment]
Thanks for the quick fix @ibhagwan ! That's super appreciated! I am a bit confused though, your first comment was an `autocmd BufEnter *` suggestion for me? Should this be still reported upstream? If the issue was fixed upstream, would you revert the https://github.com/ibhagwan/fzf-lua/commit/0babfa852ed503e66cb11970e2a0aa4b821e9cef ?

**Comment by ibhagwan** [comment]
> Thanks for the quick fix @ibhagwan ! That's super appreciated! I am a bit confused though, your first comment was an `autocmd BufEnter *` suggestion for me? Should this be still reported upstream? If the issue was fixed upstream, would you revert the [0babfa8](https://github.com/ibhagwan/fzf-lua/commit/0babfa852ed503e66cb11970e2a0aa4b821e9cef) ?

It smells like an upstream issue, the active window shouldn’t change in the combo of the autocmd and floating window open but I can’t really pinpoint the reason or the exact fault, that would require a deeper investigation which I’m not going to do :)
